mlir/lib/Dialect/Rock/Transforms/AccelEmitter.cpp

```diff
@@ -0,0 +1,431 @@
+      {splitMemoryCoords.endSize("j") / blocksInOutRegs, blocksInOutRegs});
+  toRowsAndCols.passThrough(
+      {"vec_group", "vec_item"},
+      {rowsAndColsIdxs["vec_group"], rowsAndColsIdxs["vec_item"]},
+      {"vec_group", "vec_item"});
+
+  TransformMapAttr toRowsAndColsAttr = toRowsAndCols.get();
+
+  auto toMatrixC = TopDownTMBuilder::below(toRowsAndCols, toRowsAndColsAttr);
+  toMatrixC.passThrough({"gemmG"}, {0}, {"g"});
+
+  toMatrixC.embed(
+      "gemmM", 1, M,
+      {"m", "wave_m", "m_tid", "m_i", "blk_row", "vec_group", "vec_item"},
+      {mPerBlock, mPerWave, rowGroupSize, mPerRepeat, m,
+       inputSpansPerMfmaIn * rowGroupSize, 1});
+  toMatrixC.embed("gemmN", 2, N, {"n", "wave_n", "n_i", "blk_col", "n_tid"},
+                  {nPerBlock, nPerWave, nPerRepeat, n, 1});
+  TransformMapAttr toMatrixCAttr = toMatrixC.get();
+
+  ArrayAttr idToMatrixCMaps =
+      b.getArrayAttr({splitMemoryCoordsAttr, toRowsAndColsAttr, toMatrixCAttr});
+  return idToMatrixCMaps;
+}
+
+Value MfmaEmitter::computeOutputConversion(PatternRewriter &b, Location loc,
+                                           int64_t M, int64_t N,
+                                           int64_t blockSize, int64_t gridSize,
+                                           Value regVectorOrig, Value regDest,
+                                           bool forceUnroll) {
+
+  Type destType = regDest.getType().dyn_cast<MemRefType>().getElementType();
+
+  int64_t accVectorLen = accVectorType.getNumElements();
+  int64_t numElements = accVectorLen * (mRepeats * nRepeats * nResultVectors);
+  auto zeroConstantOp = b.create<ConstantIndexOp>(loc, 0);
+
+  BottomUpTMBuilder toRegCScalar(b, {"scalar"}, {numElements}, loc);
+  toRegCScalar.embed({"vector"}, {0}, {mRepeats * nRepeats * nResultVectors},
+                     "scalar", {accVectorLen});
+  TransformMapAttr toRegCScalarAttr = toRegCScalar.get();
+
+  auto convertLoop = b.create<TransformingForOp>(
+      loc, ArrayRef<ValueRange>{{zeroConstantOp}, {zeroConstantOp}},
+      ArrayRef<Attribute>{b.getArrayAttr({}), b.getArrayAttr(toRegCScalarAttr)},
+      /*bounds=*/ArrayRef<int64_t>{mRepeats * nRepeats * nResultVectors},
+      /*strides=*/std::nullopt, forceUnroll, /*useIndexDiffs=*/true);
+  {
+    OpBuilder::InsertionGuard guard(b);
+    b.setInsertionPointToStart(convertLoop.getBody());
+    Value loaded =
+        b.create<memref::LoadOp>(loc, accVectorType, regVectorOrig,
+                                 convertLoop.getLowerCoords(/*domain*/ 0));
+    Value cast = loaded;
+    if (destType != accVectorType.getElementType()) {
+      VectorType destVectorType = accVectorType.clone(destType);
+      cast = createTypeConversionOp(b, loc, loaded, destVectorType);
+    }
+    b.create<InBoundsStoreOp>(loc, cast, regDest,
+                              convertLoop.getLowerCoords(/*domain*/ 1));
+  }
+  return regDest;
+}
+
+Value MfmaEmitter::computeLdsSourceOffset(OpBuilder &kb, Value k_i,
```

Comment:
Can we clean up these variable names? For one, `MN` -> `d` per gridwise conventions, but also, some of these could use lenghtening now that they're a meaningfully separate function